lib/CliftPipes/Headers.cpp

```diff
@@ -0,0 +1,145 @@
+  }
+};
+
+static pipeline::RegisterPipe<ModelHeaderPipe> ModelHeader;
+
+class HelperHeaderPipe {
+public:
+  static constexpr auto Name = "emit-new-helper-header";
+
+  std::array<pipeline::ContractGroup, 1> getContract() const {
+    using namespace pipeline;
+    using namespace revng::kinds;
+
+    return { ContractGroup({ Contract(CliftFunction,
+                                      0,
+                                      NewHelperHeader,
+                                      1,
+                                      InputPreservation::Preserve) }) };
+  }
+
+  void run(pipeline::ExecutionContext &EC,
+           const revng::pipes::CliftContainer &CliftContainer,
+           HelperHeaderContainer &HeaderFile) {
+    mlir::ModuleOp Module = CliftContainer.getModule();
+
+    llvm::raw_string_ostream Out = HeaderFile.asStream();
+
+    ptml::CTokenEmitter PTML(Out, ptml::Tagging::Enabled);
+
+    // TODO: select target properly
+    const auto &Target = TargetCImplementation::Default;
+    revng_assert(mlir::clift::verifyCSemantics(Module, Target).succeeded());
+    mlir::clift::emitHelperHeader(PTML, Target, Module);
+
+    Out.flush();
+
+    EC.commitUniqueTarget(HeaderFile);
+  }
+};
+
+static pipeline::RegisterPipe<HelperHeaderPipe> HelperHeader;
+
+class GenerateNewTypeDefinition {
+public:
+  static constexpr auto Name = "generate-new-type-definitions";
+
+  std::array<pipeline::ContractGroup, 1> getContract() const {
+    using namespace pipeline;
+    using namespace revng::kinds;
+
+    return { ContractGroup({ Contract(CliftModule,
+                                      0,
+                                      ModelTypeDefinition,
+                                      1,
+                                      InputPreservation::Preserve) }) };
+  }
+
+  void run(pipeline::ExecutionContext &EC,
+           const revng::pipes::CliftContainer &CliftContainer,
+           TypeDefinitionContainer &ModelTypesContainer) {
+
+    mlir::clift::TypeSystemEmitterConfiguration Configuration{
+      .EnablePrintingOfTheMaximumEnumValue = true,
+      .SupressExplicitPadding = true
+    };
+
+    const model::Binary &Model = *revng::getModelFromContext(EC);
+    mlir::ModuleOp Module = CliftContainer.getModule();
+
+    for (const model::TypeDefinition &Type :
+         revng::getTypeDefinitionsAndCommit(EC, ModelTypesContainer.name())) {
+      std::string &Result = ModelTypesContainer[Type.key()];
+      llvm::raw_string_ostream Out(Result);
+
+      ptml::CTokenEmitter PTML(Out, ptml::Tagging::Enabled);
+
+      // TODO: select target properly
+      const auto &Target = TargetCImplementation::Default;
+      revng_assert(mlir::clift::verifyCSemantics(Module, Target).succeeded());
+
+      std::optional<mlir::clift::DefinedType>
+        MaybeCliftType = mlir::clift::lookupCliftType(Module, Type);
+      revng_check(MaybeCliftType.has_value());
+
+      emitSingleTypeDefinition(PTML, Target, *MaybeCliftType, Configuration);
+
```

Comment:
I think this is another occurrence of code that should be moved into `lib/CliftImportModel`. And the same logic as the other occurrences applies to the `lookupCliftType` function`.